stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -763,6 +764,13 @@ public extension _TensorComputation {
       let joinStatus = pthread_join(pthread, nil)
       internalConsistencyCheck(joinStatus == 0)
       self.pthread = nil
+    } else {
+      internalConsistencyCheck(
```

Comment:
Is the purpose of this mainly to prevent intermediate changes to `_RuntimeConfig.usesSynchronousExecution`?

Changing this global variable at any time between Start and End should be considered inconsistency or UB. In that case, we could add something to `willSet` to crash if it's set in between the start and end of any computation.

Reply:
The sanity check here focuses on `pthread` -- we want to make sure that when `finish()` is called, `pthread` is not nil. But we make an exception for the case of sync execution, where `pthread` is can legitimately be nil here.